libtenzir/src/package.cpp

```diff
@@ -719,6 +720,41 @@ auto package::load(const std::filesystem::path& dir, diagnostic_handler& dh)
                          std::back_inserter(path), [](auto component) {
                            return std::string{component};
                          });
+          // Validate that all path segments are valid TQL identifiers:
+          // first char [A-Za-z_], subsequent [A-Za-z0-9_].
+          auto is_valid_ident = [](std::string_view s) -> bool {
+            if (s.empty())
+              return false;
+            auto is_alpha = [](unsigned char c) {
+              return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z');
+            };
+            auto is_alnum_us = [&](unsigned char c) {
+              return is_alpha(c) || (c >= '0' && c <= '9') || c == '_';
+            };
+            if (!(is_alpha(static_cast<unsigned char>(s[0])) || s[0] == '_'))
+              return false;
+            for (size_t i = 1; i < s.size(); ++i) {
+              if (!is_alnum_us(static_cast<unsigned char>(s[i])))
+                return false;
+            }
+            return true;
+          };
```

Comment:
Re-use the identifier parser. Needs to be exposed from the tokenizer.

Reply:
We also need to check the package_id. For ids with `-` chars we automatically replace them with `_` for operators.